### PR TITLE
Add Force update to edit service if service is in deployment

### PR DESCRIPTION
**Pull request by Poltergeist**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @jfurrow

File: src/js/components/modals/ServiceFormModal.js [hunk 1]
```diff
@@ -209,6 +210,7 @@ class ServiceFormModal extends mixin(StoreMixin) {
 
   onMarathonStoreServiceEditError(errorMessage) {
     this.setState({
+      force: errorMessage.message && /force=true/.test(errorMessage.message),
```
Comment (jfurrow) [inline src/js/components/modals/ServiceFormModal.js]: What do you think about putting this `/force=true/.test...` check in a separate method on the `ServiceFormModal`? Something like `#shouldForceEdit` that accepts the `message` and returns `/force=true/.test(message)`.

**Comment by jfurrow** [comment]
One minor comment, but this looks good!

File: src/js/components/modals/ServiceFormModal.js (hunk 1 above)
Comment (Poltergeist) [inline src/js/components/modals/ServiceFormModal.js]: Comment has been addressed.

File: src/js/events/MarathonActions.js [hunk 2]
```diff
@@ -89,9 +89,15 @@ module.exports = {
     });
   },
 
-  editService: function (data) {
+  editService: function (data, force) {
+    let url = `${Config.rootUrl}${Config.marathonAPIPrefix}/apps/${data.id}`;
+
+    if (force === true) {
+      url += '?force=true';
```
Comment (rcorral) [inline src/js/events/MarathonActions.js]: It would be good to add a simple unit test to make sure that the `force` argument changes the requested URL.

**Comment by MatApple** [comment]
Hey @Poltergeist can you post a screenshot of integration tests passing? After that, and Rafas comment I think this looks good!

File: src/js/components/modals/ServiceFormModal.js [hunk 3]
```diff
@@ -205,13 +206,18 @@ class ServiceFormModal extends mixin(StoreMixin) {
     });
   }
 
+  shouldForceUpdate(message = this.errorMessage) {
```
Comment (pierlo-upitup) [inline src/js/components/modals/ServiceFormModal.js]: Probably unrelated, but why aren't we storing `errorMessage` in this.state?

**Comment by Poltergeist** [comment]
image

image

Added a unit test and fixed the integration tests.

File: src/js/components/modals/ServiceFormModal.js (hunk 3 above)
Comment (Poltergeist) [inline src/js/components/modals/ServiceFormModal.js]: shouldForceUpdate is called in the setStates. This is why this function takes a parameter. But in fact the default is wrong. Changed it.

**Pull request merged**